cumulusci/cli/tests/test_org.py

```diff
@@ -143,6 +145,72 @@ def test_org_connect(self, oauth2client):
         assert org_config.expires == "Persistent"
         runtime.keychain.set_default_org.assert_called_once_with("test")
 
+    @contextmanager
+    def mock_main_context(self):
+        with mock.patch(  # side effects break other tests
+            "cumulusci.cli.cci.init_logger", mock.Mock()
+        ), mock.patch(
+            "cumulusci.cli.cci.get_tempfile_logger",
+            mock.MagicMock(return_value=(None, "")),
+        ), mock.patch(
+            "cumulusci.cli.cci.tee_stdout_stderr", mock.MagicMock()
+        ), mock.patch(
+            "cumulusci.cli.cci.CliRuntime", mock.Mock()
+        ):
+            yield
+
+    @responses.activate
+    def test_org_connect_sandbox(self):
+        def check_and_quit_function_halfway_through(auth_uri, token_uri, **kwargs):
+            assert "test.salesforce" in auth_uri
+            assert "test.salesforce" in token_uri
+            raise StopIteration("Done!")
```

Comment:
This way of testing is, I think, unusual enough to warrant a comment explaining it. Wouldn't it be simpler to refactor org_connect to use one function for setting up the client and another one for actually making use of it, so that you can simply mock the latter?

Reply:
> I can reintroduce the same bug and all of the tests pass.

@prescod - How are you reproducing the issue?